compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -709,6 +710,50 @@ declare_lint! {
     "detect assignments that will never be read"
 }
 
+declare_lint! {
+    /// The `dangerous_stack_allocation` lint detects stack allocations that
+    /// are 1 GB or more.
+    ///
+    /// ### Example
+    ///
+    /// ``` rust
+    /// fn func() {
+    ///     const CAP: usize = std::u32::MAX as usize;
+    ///     let mut x: [u8; CAP>>1] = [0; CAP>>1];
+    ///     x[2] = 123;
+    ///     println!("{}", x[2]);
+    /// }
+    ///
+    /// fn main() {
+    ///     std::thread::Builder::new()
+    ///         .stack_size(3 * 1024 * 1024 * 1024)
+    ///         .spawn(func)
+    ///         .unwrap()
+    ///         .join()
+    ///         .unwrap();
+    /// }
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ```text
+    /// warning: allocation of size: 1 GiB  exceeds most system architecture limits
+    /// --> $DIR/large-stack-size-issue-83060.rs:7:9
+    /// |
+    /// LL |     let mut x: [u8; CAP>>1] = [0; CAP>>1];
+    ///  |         ^^^^^
+    ///  |
+    ///  = note: `#[warn(dangerous_stack_allocation)]` on by default
```

Comment:
This will likely fail because of the textual formatting not being correctly left aligned.

Reply:
Thank you! I would have never guessed that.